File: android-sdk/src/main/kotlin/com/example/featureflagsdk/android/FeatureContext.kt
package com.example.featureflagsdk.android

/**
 * Evaluation context passed by the calling app.
 *
 * @param subjectKey Unique identifier for the subject being evaluated (e.g. user ID).
 * @param region     Geographic region (e.g. "Asia", "North America").
 * @param subject    Subject group (e.g. "vip", "internal").
 * @param releaseKey Release tag (e.g. "20260518").
 * @param attributes Additional free-form attributes forwarded to the backend.
 */
data class FeatureContext(
    val subjectKey: String,
    val region: String? = null,
    val subject: String? = null,
    val releaseKey: String? = null,
    val attributes: Map<String, String> = emptyMap(),
)

File: ios-sdk/Sources/FeatureFlagSDK/FeatureContext.swift
import Foundation

/// Evaluation context passed by the calling app.
public struct FeatureContext {
    /// Unique identifier for the subject being evaluated (e.g. user ID).
    public let subjectKey: String
    /// Geographic region (e.g. "Asia", "North America").
    public let region: String?
    /// Subject group (e.g. "vip", "internal").
    public let subject: String?
    /// Release tag (e.g. "20260518").
    public let releaseKey: String?
    /// Additional free-form attributes forwarded to the backend.
    public let attributes: [String: String]

    public init(
        subjectKey: String,
        region: String? = nil,
        subject: String? = nil,
        releaseKey: String? = nil,
        attributes: [String: String] = [:]
    ) {
        self.subjectKey = subjectKey
        self.region = region
        self.subject = subject
        self.releaseKey = releaseKey
        self.attributes = attributes
    }
}
